### PR TITLE
Add action for spark 3 scala 2.12

### PR DESCRIPTION
**Pull request by Zejnilovic**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -36,10 +36,29 @@ jobs:
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-2_12-3_0-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-2_12-3_0-
+      - name: Switch scala to 2_11
```
Comment (yruslan) [inline .github/workflows/build.yml:61]:
```suggestion
      - name: Switch scala to 2_12
```

**Comment by Zejnilovic** [comment]
Examples fail on file not there. Even locally. But the file exists. Will evaluate.

**Comment by Zejnilovic** [comment]
This is gonna take some time on internal meetings. Writing in spark is non-blocking and thus the assert for the existence of a file is faster than the actual write. Looking for better solutions than `Thread.sleep`

**Review by yruslan — APPROVED** [review]

**Review by yruslan — APPROVED** [review]
Checked the code

**Review by dk1844 — APPROVED** [review]
LGTM. It seems that the after-Spark-write-check was a pure trap, sorry about that. (just read the code)

**Pull request merged**